main/githooks.py

```diff
@@ -350,6 +360,27 @@ def trim_trailing_whitespace_in_file(filename, new_file, in_place):
     return 0
 
 
+class TestTrimTrailingWhitespace(unittest.TestCase):
+    def test_trim_trailing_whitespace(self):
+        test_file = Path.cwd() / 'tttw_testfile.txt'
```

Comment:
cwd might not be what you expect as I've been finding out lately :)
Could you use a context manager from tempfile please?

Reply:
Good point. Done.